lib/middleware/pronto.rb

```diff
@@ -0,0 +1,32 @@
+class Pronto
+  class PathMatcher
+    PATTERN = %r{^((?:/a/)(?<slug>[\w-]+)(?:/?)(?:(\?.*)*))$}.freeze
+
+    def self.match(path)
+      match = PATTERN.match(path)
+      return match[:slug] if match
+    end
+  end
+
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    @status, @headers, @response = @app.call(env)
+    req = Rack::Request.new(env)
+
+    path_match = PathMatcher.match(req.path)
+
+    if path_match
+      page = Page.find_by(slug: path_match)
+
+      if page.language_code == 'en' && page.petition_page?
```

Comment:
Don't we need to add a null check for `page` here? In case the requested slug doesn't exist on our DB:
`/a/this-page-does-not-exist`
```suggestion
      if page&.language_code == 'en' && page&.petition_page?
```